### PR TITLE
resolves issue #200 enables webhook commands on subgroups

### PR DESCRIPTION
**Pull request by ericjaystevens**
#### Summary

Enables support for managing group scopes webhooks in subgroups and project scope webhooks in projects that are inside a subgroup.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-plugin-gitlab/issues/200

### CODE REVIEW

File: server/command.go [hunk 1]
```diff
@@ -242,15 +242,15 @@ func (p *Plugin) webhookCommand(parameters []string, info *gitlab.GitlabUserInfo
 		}
 
 		namespace := parameters[1]
-		fullPath := strings.Split(namespace, "/")
+		group, project, namespaceErr := p.GitlabClient.ResolveNamespaceAndProject(info, namespace, enablePrivateRepo)
+		if namespaceErr != nil {
```
Comment (hanzei) [inline server/command.go:246]: Nit: Should this name just be `err` as it's idiomatic?

File: server/command.go [hunk 2]
```diff
@@ -305,29 +305,26 @@ func (p *Plugin) webhookCommand(parameters []string, info *gitlab.GitlabUserInfo
...
-			project, _ := p.GitlabClient.GetProject(info, owner, repo)
+		if projectName != "" {
+			project, _ := p.GitlabClient.GetProject(info, group, projectName)
 			newWebhook, err := p.GitlabClient.NewProjectHook(info, project.ID, hookOptions)
 			if err != nil {
 				return err.Error()
 			}
 			return fmt.Sprintf("Webhook Created:\n%s", newWebhook.String())
 		}
-		// If webhook is group scoped
```
Comment (hanzei) [inline server/command.go:320]: Nit: Maybe keep this comment instead of adding a new one? #reducethediff

**Review by hanzei — APPROVED** [review]
Thanks for the fix :+1: 

Two suggestions, but nothing blocking.

**Comment by hanzei** [comment]
@larkox Could you please review this PR while @iomodo is OOO?

File: server/command.go [hunk 3]
```diff
@@ -305,29 +305,26 @@ func (p *Plugin) webhookCommand(parameters []string, info *gitlab.GitlabUserInfo
...
+			return namespaceErr.Error()
+		}
 		//if project scope
-		if len(fullPath) == 2 {
-			owner := fullPath[0]
-			repo := fullPath[1]
-
-			project, _ := p.GitlabClient.GetProject(info, owner, repo)
+		if projectName != "" {
+			project, _ := p.GitlabClient.GetProject(info, group, projectName)
```
Comment (larkox) [inline server/command.go:315]: Why are we not checking this error?

**Review by larkox — CHANGES_REQUESTED** [review]
LGTM, but I want to make sure about a small detail.

**Comment by storrgie** [comment]
Is possible to get some love on this? We've got a pretty deep structure in gitlab that we want to use this bot with. We're new licensees of both gitlab ultimate and mattermost e20.

**Comment by hanzei** [comment]
@ericjaystevens Would you mind adding the error check for https://github.com/mattermost/mattermost-plugin-gitlab/pull/202#discussion_r492654104? Let me know if you need help.

**Review by hanzei — APPROVED** [review]

**Review by larkox — APPROVED** [review]
LGTM, thanks!

**Comment by storrgie** [comment]
Again tapping to see if this can get some love. We can't use the plugin with gitlab at this time (all of our repositories are located 2+ levels down in organizational structure).

**Comment by jasonblais** [comment]
@DHaussermann kind reminder to help with QA review, thanks! :)

**Review by DHaussermann — COMMENTED** [review]
Tested and passed
- Confirmed that creating a webhook with the slash command works as expected within a sub-group
- Brief regression testing to ensure `private` flag of the project is respected and event delivery is working 
- No issues found

LGTM!

Added release test to cover this sub-group case.

Thanks @ericjaystevens for addressing this. Much appreciated!

**Review by DHaussermann — APPROVED** [review]

**Pull request merged**